library.rb

```diff
@@ -0,0 +1,174 @@
+require_relative './book'
+require_relative './person'
+require_relative './teacher'
+require_relative './student'
+require_relative './rental'
+
+class Library
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      print "Library is empty\n\n"
+      return
+    end
+    @books.each do |book|
+      print "Title: #{book.title.capitalize}, Author: #{book.author.capitalize}\n"
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      print "No person added\n\n"
+      return
+    end
+    @people.each do |person|
+      print "[#{person.class.name}] Name: #{person.name.capitalize}, ID: #{person.id}, Age: #{person.age}\n"
+    end
+  end
+
+  def list_all_rental
+    if @rentals.empty?
+      print "No rentals exist\n\n"
+      return
+    end
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    person = @people[id]
+    puts person.name
+    puts 'Rentals: '
+    all_rentals = person.rentals
+    puts 'No rental found' if all_rentals.empty?
+    all_rentals.each do |rental|
+      print "Date: #{rental.date}, Book \'#{rental.book.title}\' by #{rental.book.author}\n"
+    end
```

Comment:
- [ ] I created a `book` and a `person` and created a `rental` which was created successfully. However, when I tried to list all `rentals` the app crashes. I believe this was caused by the variable `person` which you created from the people array. The problem is that you are trying to get the index of the person from the people array by using the `id` as the index which I believe is not possible because their `indexes` are quite different from their `ids`. Below is what I did in your app to resolve this error 😄.

```suggestion
    rentals = @rentals.filter {|rental| rental.person.id == id}
    puts "Rentals:"
    rentals.each do |rental|
      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}"
    end
```


Here is the error I was getting before I resolved it:
<img width="580" alt="Loom_oYV7gJlGtd" src="https://user-images.githubusercontent.com/81468675/153689707-4d784a9c-738e-48ac-a03d-8dd76b8d2919.png">